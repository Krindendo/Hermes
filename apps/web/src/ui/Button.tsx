import React, { ButtonHTMLAttributes, DetailedHTMLProps } from "react";
import styled, { css } from "styled-components";

export type ButtonProps = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> & {
  size?: keyof typeof sizeClassnames;
  color?: keyof typeof colorClassnames;
};

const Button: React.FC<ButtonProps> = ({ children, disabled, ref, ...props }) => (
  <Container data-testid="button" {...props}>
    {children}
  </Container>
);

const sizeClassnames = {
  big: css`
    padding: 10px;
    border-radius: 10px;
  `,
  small: css``,
  tiny: css``,
};

const colorClassnames = {
  primary: css`
    background-color: ${({ theme }) => theme.palette.primary.main};
  `,
  secondary: css``,
  error: css``,
  transparent: css``,
};

const Container = styled.button<ButtonProps>`
  cursor: pointer;

  ${(props) => sizeClassnames[props.size || "big"]}
  ${(props) => colorClassnames[props.color || "primary"]}
`;

export default Button;
