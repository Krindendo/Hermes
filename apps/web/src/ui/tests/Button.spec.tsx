import React from "react";

import { Button } from "../Button";

describe("Button", () => {
  it("shoude render a button with Click me text", () => {});

  it("should render a disabled button if disabled prop is passed", () => {});

  it("should render an icon if icon props is passed", () => {});

  it("should match snapshot", () => {});
});
