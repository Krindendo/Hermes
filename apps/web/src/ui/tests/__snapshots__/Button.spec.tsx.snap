exports[`Button should match snapshot 1`] = ``;